state/factory/workingset.go

```diff
@@ -194,7 +194,7 @@ func (ws *workingSet) runAction(
 	}
 	// for replay tx, check against deployer whitelist
 	g := genesis.MustExtractGenesisContext(ctx)
-	if selp.Encoding() == uint32(iotextypes.Encoding_ETHEREUM_UNPROTECTED) && !g.IsDeployerWhitelisted(selp.SenderAddress()) {
+	if !selp.Protected() && !g.IsDeployerWhitelisted(selp.SenderAddress()) {
```

Comment:
The behavior has not actually changed because the `encoding` here will not be the `Encoding_TX_CONTAINER`, so a hardfork is not needed.